### ISSUE TITLE
NoMethodError: undefined method `downcase' for nil:NilClass

### ISSUE THREAD

**Issue by Viajaz** [pglombardo/PasswordPusher#425]
Occurred when trying to use the `latest`, `1.17.5` and `1.17.4` tagged docker images, had to rollback to `release`.
```
NoMethodError: undefined method `downcase' for nil:NilClass

  config.log_level = Settings.log_level.downcase.to_sym
                                       ^^^^^^^^^
/opt/PasswordPusher/config/environments/production.rb:55:in `block in <main>'
/opt/PasswordPusher/vendor/bundle/ruby/3.1.0/gems/railties-6.1.7/lib/rails/railtie.rb:234:in `instance_eval'
/opt/PasswordPusher/vendor/bundle/ruby/3.1.0/gems/railties-6.1.7/lib/rails/railtie.rb:234:in `configure'
/opt/PasswordPusher/config/environments/production.rb:1:in `<main>'
/opt/PasswordPusher/vendor/bundle/ruby/3.1.0/gems/bootsnap-1.13.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:32:in `require'
/opt/PasswordPusher/vendor/bundle/ruby/3.1.0/gems/bootsnap-1.13.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:32:in `require'
/opt/PasswordPusher/vendor/bundle/ruby/3.1.0/gems/zeitwerk-2.6.1/lib/zeitwerk/kernel.rb:35:in `require'
/opt/PasswordPusher/vendor/bundle/ruby/3.1.0/gems/activesupport-6.1.7/lib/active_support/dependencies.rb:332:in `block in require'
/opt/PasswordPusher/vendor/bundle/ruby/3.1.0/gems/activesupport-6.1.7/lib/active_support/dependencies.rb:299:in `load_dependency'
/opt/PasswordPusher/vendor/bundle/ruby/3.1.0/gems/activesupport-6.1.7/lib/active_support/dependencies.rb:332:in `require'
/opt/PasswordPusher/vendor/bundle/ruby/3.1.0/gems/railties-6.1.7/lib/rails/engine.rb:571:in `block (2 levels) in <class:Engine>'
/opt/PasswordPusher/vendor/bundle/ruby/3.1.0/gems/railties-6.1.7/lib/rails/engine.rb:570:in `each'
/opt/PasswordPusher/vendor/bundle/ruby/3.1.0/gems/railties-6.1.7/lib/rails/engine.rb:570:in `block in <class:Engine>'
/opt/PasswordPusher/vendor/bundle/ruby/3.1.0/gems/railties-6.1.7/lib/rails/initializable.rb:32:in `instance_exec'
/opt/PasswordPusher/vendor/bundle/ruby/3.1.0/gems/railties-6.1.7/lib/rails/initializable.rb:32:in `run'
/opt/PasswordPusher/vendor/bundle/ruby/3.1.0/gems/railties-6.1.7/lib/rails/initializable.rb:61:in `block in run_initializers'
/opt/PasswordPusher/vendor/bundle/ruby/3.1.0/gems/railties-6.1.7/lib/rails/initializable.rb:50:in `each'
/opt/PasswordPusher/vendor/bundle/ruby/3.1.0/gems/railties-6.1.7/lib/rails/initializable.rb:50:in `tsort_each_child'
/opt/PasswordPusher/vendor/bundle/ruby/3.1.0/gems/railties-6.1.7/lib/rails/initializable.rb:60:in `run_initializers'
/opt/PasswordPusher/vendor/bundle/ruby/3.1.0/gems/railties-6.1.7/lib/rails/application.rb:391:in `initialize!'
/opt/PasswordPusher/config/environment.rb:78:in `<main>'
/opt/PasswordPusher/vendor/bundle/ruby/3.1.0/gems/bootsnap-1.13.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:32:in `require'
/opt/PasswordPusher/vendor/bundle/ruby/3.1.0/gems/bootsnap-1.13.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:32:in `require'
/opt/PasswordPusher/vendor/bundle/ruby/3.1.0/gems/zeitwerk-2.6.1/lib/zeitwerk/kernel.rb:35:in `require'
/opt/PasswordPusher/vendor/bundle/ruby/3.1.0/gems/activesupport-6.1.7/lib/active_support/dependencies.rb:332:in `block in require'
/opt/PasswordPusher/vendor/bundle/ruby/3.1.0/gems/activesupport-6.1.7/lib/active_support/dependencies.rb:299:in `load_dependency'
/opt/PasswordPusher/vendor/bundle/ruby/3.1.0/gems/activesupport-6.1.7/lib/active_support/dependencies.rb:332:in `require'
/opt/PasswordPusher/vendor/bundle/ruby/3.1.0/gems/railties-6.1.7/lib/rails/application.rb:367:in `require_environment!'
/opt/PasswordPusher/vendor/bundle/ruby/3.1.0/gems/railties-6.1.7/lib/rails/application.rb:533:in `block in run_tasks_blocks'
/opt/PasswordPusher/vendor/bundle/ruby/3.1.0/gems/rake-13.0.6/exe/rake:27:in `<top (required)>'
/opt/PasswordPusher/vendor/bundle/ruby/3.1.0/gems/bundler-2.3.3/lib/bundler/cli/exec.rb:58:in `load'
/opt/PasswordPusher/vendor/bundle/ruby/3.1.0/gems/bundler-2.3.3/lib/bundler/cli/exec.rb:58:in `kernel_load'
/opt/PasswordPusher/vendor/bundle/ruby/3.1.0/gems/bundler-2.3.3/lib/bundler/cli/exec.rb:23:in `run'
/opt/PasswordPusher/vendor/bundle/ruby/3.1.0/gems/bundler-2.3.3/lib/bundler/cli.rb:484:in `exec'
/opt/PasswordPusher/vendor/bundle/ruby/3.1.0/gems/bundler-2.3.3/lib/bundler/vendor/thor/lib/thor/command.rb:27:in `run'
/opt/PasswordPusher/vendor/bundle/ruby/3.1.0/gems/bundler-2.3.3/lib/bundler/vendor/thor/lib/thor/invocation.rb:127:in `invoke_command'
/opt/PasswordPusher/vendor/bundle/ruby/3.1.0/gems/bundler-2.3.3/lib/bundler/vendor/thor/lib/thor.rb:392:in `dispatch'
/opt/PasswordPusher/vendor/bundle/ruby/3.1.0/gems/bundler-2.3.3/lib/bundler/cli.rb:31:in `dispatch'
/opt/PasswordPusher/vendor/bundle/ruby/3.1.0/gems/bundler-2.3.3/lib/bundler/vendor/thor/lib/thor/base.rb:485:in `start'
/opt/PasswordPusher/vendor/bundle/ruby/3.1.0/gems/bundler-2.3.3/lib/bundler/cli.rb:25:in `start'
/opt/PasswordPusher/vendor/bundle/ruby/3.1.0/gems/bundler-2.3.3/exe/bundle:48:in `block in <top (required)>'
/opt/PasswordPusher/vendor/bundle/ruby/3.1.0/gems/bundler-2.3.3/lib/bundler/friendly_errors.rb:103:in `with_friendly_errors'
/opt/PasswordPusher/vendor/bundle/ruby/3.1.0/gems/bundler-2.3.3/exe/bundle:36:in `<top (required)>'
/usr/local/bundle/bin/bundle:25:in `load'
/usr/local/bundle/bin/bundle:25:in `<main>'
Tasks: TOP => db:migrate => db:load_config => environment
(See full trace by running task with --trace)
```

**Comment by pglombardo** [comment]
If you want a quick fix, add `Settings.log_level = 'warn'` to your settings.yml file.

I'm looking for a better method to handling missing values.  I'll have a fix out soon.

**Comment by Viajaz** [comment]
Is this actually #400? Should this be closed as duplicate?
@pglombardo

**Comment by pglombardo** [comment]
I don't mind duplicates.  All the more pressure to remind me that I have to fix them. :-)

**Comment by pglombardo** [comment]
The fix for this has been merged and is building now in container v1.17.6.
